### ISSUE TITLE
Release config should reside in  /config/release.exs folder

### ISSUE THREAD

**Issue by sikanhe** [bitwalker/distillery#93]
It really does not make sense right now for release config to reside in rel. When I think about rel i think about files that are produced by the release (similar to _build), not something I should version control. 

I understand that in upcoming version of distillery the release will be in _build, which makes this problem even more apparent. 

Is there any reason why release config cannot be in the same folder as all other config files?

**Comment by bitwalker** [comment]
The reason release output is moving to `_build` is because the real "artifacts" of a release build are the tarballs, which will remain under `rel` - the workspace generated under `rel` today tends to cause issues when people try to run their release from there, and more often than not, it's not used. One will still be able to run their release from `_build` with `mix release.run` or something like that, but that's more or less the gist. I'm not clear on why it makes this any more of an issue though.

I discussed this with José previously, and after some back and forth we agreed that keeping the release configuration under `rel` makes more sense, as release and other compile-time only configuration do not belong in the same place. `config` is intended only for runtime configuration files for your application, while other compile-time configs (such as `mix.exs`, `rel/config.exs`, `rebar.config`, and others) are stored separately. It is a common situation to provide other release-related files as well, such as a custom `sys.config` or `vm.args`, and those would not make any sense to store under `config` - instead, `rel` is intended to be a place where all release-related files and artifacts are stored and source-controlled.

At least for now, it would take a pretty compelling argument for me to decide to change this. I do think it's a mistake to dump the release output to `rel`, which is why I've decided to move it to `_build`, but beyond that I still believe it's best to keep those concerns separate.
